man/timedated.conf.xml

```diff
@@ -0,0 +1,107 @@
+-->
+
+<refentry id="timedated.conf" conditional='ENABLE_TIMEDATED'
+    xmlns:xi="http://www.w3.org/2001/XInclude">
+  <refentryinfo>
+    <title>timedated.conf</title>
+    <productname>systemd</productname>
+
+    <authorgroup>
+      <author>
+        <contrib>Developer</contrib>
+        <firstname>Yu</firstname>
+        <surname>Watanabe</surname>
+        <email>watanabe.yu@gmail.com</email>
+      </author>
+    </authorgroup>
+  </refentryinfo>
+
+  <refmeta>
+    <refentrytitle>timedated.conf</refentrytitle>
+    <manvolnum>5</manvolnum>
+  </refmeta>
+
+  <refnamediv>
+    <refname>timedated.conf</refname>
+    <refname>timedated.conf.d</refname>
+    <refpurpose>Configuration files for time and date bus mechanism</refpurpose>
+  </refnamediv>
+
+  <refsynopsisdiv>
+    <para><filename>/etc/systemd/timedated.conf</filename></para>
+    <para><filename>/etc/systemd/timedated.conf.d/*.conf</filename></para>
+    <para><filename>/run/systemd/timedated.conf.d/*.conf</filename></para>
+    <para><filename>/usr/lib/systemd/timedated.conf.d/*.conf</filename></para>
+  </refsynopsisdiv>
+
+  <refsect1>
+    <title>Description</title>
+
+    <para>These configuration files control time and date bus mechanism.</para>
+
+  </refsect1>
+
+  <xi:include href="standard-conf.xml" xpointer="main-conf" />
+
+  <refsect1>
+    <title>Options</title>
+
+    <para>The following settings are configured in the <literal>[Timedate]</literal> section:</para>
+
+    <variablelist class='network-directives'>
+
+      <varlistentry>
+        <term><varname>NTPServices=</varname></term>
+        <listitem><para>A space-separated list of unit names of NTP services that are monitored and controlled
+        by systemd-timedated. When the empty string is assigned, the list of NTP services is reset, and all
+        assignments prior to this one will have no effect. This setting defaults to an empty list.</para>
+        <para>The command <command>timedatectl set-ntp on</command> enables and starts the first existing service
+        listed in this setting.</para>
+        <para>For distro packagers, it is recommended that each NTP client package adds drop-in configuration
+        file at <filename>/usr/lib/timedated.conf.d</filename>,
+        e.g., <filename>/usr/lib/timedated.conf.d/40-ntpd.conf</filename>.
+        If systemd-timesyncd is enabled to build, then the service is listed by <filename>50-timesyncd.service</filename>.
+        So, if an alternative NTP client service is prefered on a distribution, then the configuration name
+        which lists the service should have smaller prefix digits.</para>
+        <para>For administrators, if multiple NTP clients are installed in your system, then you can override
+        the order of the preference of the NTP clients by manually setting this in
+        <filename>/etc/systemd/timedated.conf</filename>:
+        <programlisting>NTPServices=ntpd chronyd systemd-timesyncd</programlisting>
```

Comment:
Based on the above, it might be preferable to include a more realistic sysadmin preference

    NTPServices=
    NTPServices=ntpd

... ah, hmm.  Given that we already have the normal priority system (for free) - are we really really sure we want the complexity of a precedence _list_?

If the config file is included in the service's package, then it can assume the service file exists, and set timedated to prefer its own service (unless there is a higher priority one or etc).  I assume systemd-timesyncd's priority is 50, and any ntp client which includes a drop-in would set a lower prefix, and ideally all the non-systemd ntp client packages would use the same priority.

If the config file is included in a centralized package for a Fedora spin, is it not good enough to for it to set the preferred service to chronyd?

I think changing to a simpler `NTPService=` (no list) could make the limitation above feel more consistent...

... except, perhaps, for the part where it will make higher prefix files "win" in `timedated.conf.d/`, v.s. system-presets/ where lower prefix files "win"?  So we would need to take care to be as clear as possible.  But arguably the list behaviour is the more complex one.  I'm not sure there's a direct connection between the way the list behaviour has to work, and the way precedence works for systemd preset files, I think it might just have been co-incidence.

Reply:
It is not necessary to clear the list by an admin when he/she writes the list in `/etc/systemd/timedated.conf`. As the distro-provided drop-in lists are added at the tail of the admin defined list.
E.g., `NTPServices=ntpd chronyd` in timedated.conf, and `NTPServices=systemd-timesyncd` in drop-in 50-timesyncd.conf, then it results `NTPServices=ntpd chronyd systemd-timesyncd`. Thus, if at least one of ntpd or chronyd is installed, then systemd-timesyncd is not enabled by `timedatectl set-ntp on`.

Of course, admin can enable specific NTP clients and completely disable the others by creating a highest-number drop-in, e.g., 99-hoge.conf with
```
NTPServices=
NTPServices=ntpd chronyd
```
In this case, systemd-timesyncd is never controlled by `timedatectl` command.

As usual, the similar effect can be gained by creating a symbolic link to /dev/null at /etc/systemd/timedated.conf.d/50-timesyncd.conf. I think this is preferable way to disable a specific NTP client.